zapisy/apps/defects/views.py

```diff
@@ -0,0 +1,297 @@
+    formset = DefectImageFormSet()
+    context = {'form': form, 'formset': formset, "response": request.method,
+               'extra_images_number': ExtraImagesNumber, "edit": edit}
+    return render(request, 'editDefect.html', context)
+
+
+def edit_defect_post_request(request, defect_id):
+    form = DefectForm(request.POST, request.FILES)
+
+    if not form.is_valid():
+        return return_error_and_reload(request, form, True, str(form.errors))
+
+    form_data = form.cleaned_data
+    defect = Defect.objects.filter(pk=defect_id)
+    formset = DefectImageFormSet(request.POST, request.FILES, instance=defect.get())
+
+    if not formset.is_valid():
+        return return_error_and_reload(request, form, True, str(formset.errors))
+
+    defect.update(name=form_data['name'], last_modification=now(),
+                  description=form_data['description'], place=form_data['place'])
+    formset.save()
+    selected_images = request.POST.getlist('files-to-delete[]')
+    for image in selected_images:
+        do_delete_image(request, image)
+
+    if request.user.id != defect.get().reporter.id:
+        defect_modified.send_robust(
+            sender=Defect,
+            instance=defect.get(),
+            user=defect.get().reporter,
+            executor=request.user
+        )
+
+    messages.success(request, "Edytowano usterkę")
+    return redirect('defects:show_defect', defect_id=defect_id)
+
+
+def add_defect_post_request(request):
+    form = DefectForm(request.POST, request.FILES)
+    if not form.is_valid():
+        return return_error_and_reload(request, form, False, str(form.errors))
+
+    creation_date = now()
+    form_data = form.cleaned_data
+    defect = Defect(name=form_data['name'], creation_date=creation_date, last_modification=creation_date,
+                    place=form_data['place'], description=form_data['description'], reporter=request.user,
+                    state=0)
+
+    formset = DefectImageFormSet(request.POST, request.FILES, instance=defect)
+    if not formset.is_valid():
+        return return_error_and_reload(request, form, False, str(formset.errors))
+
+    defect.save()
+    try:
+        formset.save()
+    except Exception as exception:
+        messages.error(request, "Wystąpił problem podczas zapisu zdjęć. Niektóre z nich mogły zostać niedodane.")
+        logging.getLogger().error("Error during uploading files to GoogleDrive: {}".format(str(exception)))
+        return redirect('defects:edit_defect', defect_id=defect.id)
+
+    messages.success(request, "Dodano pomyślnie usterkę")
+    return redirect('defects:main')
+
+
+@employee_required
+def print_defects(request):
+    defects_list = parse_names(request.GET, "defects_ids")
+    if defects_list is None or defects_list == []:
+        return render(request, 'defectsPrint.html', {'defects': Defect.objects.all()})
+    else:
+        defects = dict([(defect.id, defect) for defect in Defect.objects.filter(pk__in=defects_list)])
```

Comment:
Nie wiem, jak przegapiłem, że tutaj trzeba użyć _dictionary comprehension_.
```suggestion
        defects = { defect.id: defect for defect in Defect.objects.filter(pk__in=defects_list) }
```